### PR TITLE
Use default location lat lon for /map/

### LINKED ISSUE #7045: Fix /map/ when no layers are specified
- Fix route error so /map/ displays the same as /map/:layernames
- If lat and lon isn't specified in the params, display the user's profile location. If the user has no profile location, show a default lat and lon.

This is part of https://github.com/publiclab/plots2/issues/6993

### PR DESCRIPTION
**Pull request by nstjean**
Fixes #7045  (<=== Add issue number here)

- [x] Fix route error so /map/ displays the same as /map/:layernames
- [x] If lat and lon isn't specified in the params, display the user's profile location. If the user has no profile location, show a default lat and lon.

### CODE REVIEW

**Comment by nstjean** [comment]
When logged in with a profile location saved:
🎈 Public Lab: Map - Google Chrome_006

When not logged in / no profile location:
🎈 Public Lab: Map - Google Chrome_005

**Comment by nstjean** [comment]
I discovered that if a lat or lon of 0 was passed in via the params or by a user's profile that it was evaluating as false and using the default value instead. I have corrected this.

File: app/views/map/map.html.erb [hunk 1]
```diff
@@ -28,9 +28,21 @@
         new L.LatLng(-54.36775852, 178.59375)
       );
 
-      var lat = parseFloat(params.lat)|| 23;
-      var lon = parseFloat(params.lon) || 77;
-      var zoom = parseFloat(params.zoom) || 4;
+      var user_lat = null;
```
Comment (sagarpreet-chadha) [inline app/views/map/map.html.erb]: Hey, i think we should do all these calculations in the controller. From the controller we should pass only lat, lon, zoom, and layer_name. Thanks!

Comment (nstjean) [reply app/views/map/map.html.erb]: Ahhh, that's a good point. Can the controller access the urlHash though? It's in Javascript. I'll take a look at it all and see what I can move to the controller!

Comment (jywarren) [reply app/views/map/map.html.erb]: I think we can't... Without reimplementing urlhash in Ruby w request.fullpath maybe? Do you think it's worth it? What are the pros and cons?

Comment (nstjean) [reply app/views/map/map.html.erb]: My thought is that the Controller is only called once on page load, and the url hash is accessed many times in the map. It makes sense to keep the url hash calculations all on one page.

I do see some other things that can be cleaned up and moved to the controller though!

**Comment by nstjean** [comment]
I made the clean up changes in #7061

**Comment by jywarren** [comment]
Should this be merged now? Before or after #7061? Thanks!!!!

**Comment by nstjean** [comment]
@jywarren  Before, thank you!!!

**Comment by jywarren** [comment]
:-) all good i guess! Thanks!

**Pull request merged**